src/pusher.interceptor.ts

```diff
@@ -73,7 +73,7 @@ export class PusherInterceptor implements NestInterceptor {
 
         let channelName = channelMetadata
         //If its a channel builder then we need to invoke it
-        if (channelMetadata['req'] && channelMetadata['event']) {
+        if (typeof channelMetadata === 'function') {
```

Comment:
@underfisk This check for channelMetadata being a function was brittle and it broke for me as I used a different name for the second parameter – `eventName` instead of `event`. So, I made this change to simply check the `typeof` channelMetadata.

Reply:
@gulsharan I do believe something has changed now with v5 also but thanks for reporting this. One of the things i would ask you is if you could possible write some unit tests for assert this change so that we can be sure that its not going to break any other existing use cases

Reply:
Sure, I will add some tests. And let me also test the library with the previous version of Pusher.

Reply:
@underfisk I have added tests for all 4 decorators:

```javascript
 PASS  src/__tests__/pusher.interceptor.spec.ts
  PusherInterceptor
    @PusherChannel only
      ✓ should not send message if method is not decorated with PusherEvent (239 ms)
    @PusherChannel + @PusherEvent
      ✓ should send message if event name is available (10 ms)
      ✓ should retrieve the channel name from builder (10 ms)
    @PusherSocketId
      ✓ should use custom socketId header if provided (8 ms)
      ✓ should fall back on default socketId header (8 ms)
    @PusherSendGuard
      ✓ should send message if send guard returns true (6 ms)
      ✓ should not send message if send guard returns false (7 ms)
```

Reply:
@gulsharan Thanks for this, I'll merge and release a new version

Reply:
@gulsharan Check out the new version

Reply:
v0.0.8 works like a charm! Thanks. :)

Reply:
Let me know if I can be of any help in your other projects  – just sent you a connect on twitter.

Reply:
@gulsharan Sure, thanks for helping with this project, i haven't been using it for a while but i'm glad someone is!